### PR TITLE
Run Container Matrix Tests with a given Version on the Cmdline

### PR DESCRIPTION
**Pull request by janheise**
## Description

Adds property `graylog.matrix.tests.search.version.override` so you can run the matrix tests with `mvn verify -Dgraylog.matrix.tests.version.override=OpenSearch:2.6.0`  for a given version.

It excludes all incompatible tests (e.g. if you test for OpenSearch2, it removes all tests for Elasticsearch7 and OpenSearch1.3) and also reduces the matrix to only include the version set on the commandline.

## Motivation and Context

Can be used to test if new OpenSearch versions that are released are still compatible without having to change our code to include the new version.

Can be used to split the IT tests into several runs to reduce runtime

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/test/java/org/graylog/testing/containermatrix/ContainerMatrixTestEngine.java [hunk 1]
```diff
@@ -98,12 +99,57 @@ private Set<Class<? extends PluginJarsProvider>> getPluginJarsProvider(Set<Class
...
     }
 
-    private Set<SearchServer> getSearchServerVersions(Set<Class<?>> annotatedClasses) {
+    /**
+     * Property has to be in the form of Elasticsearch|OpenSearch-x.x.x
+     *
+     * @return
+     */
+    private static String getVersionOverride() {
+        return System.getProperty("graylog.matrix.tests.version.override");
```
Comment (bernd) [inline graylog2-server/src/test/java/org/graylog/testing/containermatrix/ContainerMatrixTestEngine.java:108]: How about making the property name "search version" specific? If we want to do the same for MongoDB at some point, we would have a naming problem. :smile: 

```suggestion
        return System.getProperty("graylog.matrix.tests.search.version.override");
```

File: graylog2-server/src/main/java/org/graylog2/storage/SearchVersion.java [hunk 2]
```diff
@@ -43,6 +43,15 @@ public enum Distribution {
         public String toString() {
             return this.printName;
         }
+
+        public static Distribution valueOfLabel(String label) {
```
Comment (todvora) [inline graylog2-server/src/main/java/org/graylog2/storage/SearchVersion.java:47]: This part seems very similar to SearchVersion#decode. Could we reuse it here?

File: graylog2-server/src/test/java/org/graylog/testing/containermatrix/ContainerMatrixTestEngine.java [hunk 3]
```diff
@@ -98,12 +99,57 @@ private Set<Class<? extends PluginJarsProvider>> getPluginJarsProvider(Set<Class
...
     }
 
-    private Set<SearchServer> getSearchServerVersions(Set<Class<?>> annotatedClasses) {
+    /**
+     * Property has to be in the form of Elasticsearch|OpenSearch-x.x.x
+     *
+     * @return
+     */
+    private static String getSearchVersionOverride() {
+        return System.getProperty("graylog.matrix.tests.search.version.override");
```
Comment (todvora) [inline graylog2-server/src/test/java/org/graylog/testing/containermatrix/ContainerMatrixTestEngine.java:108]: What about doing all the splitting, parsing and comparisions here and return Optional<SearchVersion> that could be then used elsewhere?

**Review by todvora — APPROVED** [review]
LGTM

**Pull request merged**